src/platform/posix/tcp_ip_channel.c

```diff
@@ -365,15 +375,20 @@ lf_ret_t TcpIpChannel_receive(NetworkChannel *untyped_self, FederateMessage *ret
       case ECONNRESET:
       case ENOTCONN:
       case ECONNABORTED:
-        self->state = NETWORK_CHANNEL_STATE_LOST_CONNECTION;
-        return LF_CONNECTION_CLOSED;
+        _update_state(self, NETWORK_CHANNEL_STATE_LOST_CONNECTION);
+        return LF_ERR;
         break;
       }
       continue;
     } else if (bytes_read == 0) {
       // This means the connection was closed.
-      self->state = NETWORK_CHANNEL_STATE_LOST_CONNECTION;
-      return LF_CONNECTION_CLOSED;
+      LF_INFO(NET, "Connection closed. Setting last known tag to FOREVER for all input ports");
+      if (self->federated_connection) {
+        self->federated_connection->network_channel_state_changed(self->federated_connection);
```

Comment:
If I remember correctly, we talked about removing this call altogether and rely on the runtime checking the network channel state before waiting for input ports to become known at a tag.